src/System.Management.Automation/engine/SessionStateContainer.cs

```diff
@@ -2382,16 +2382,38 @@ private bool HasGetChildItemDynamicParameters(ProviderInfo providerInfo)
                         // Since the path contained glob characters or we are recursing and the
```

Comment:
@iSazonov, your last commit had 21 failures in `PowerShell-CI-macos`
(These are 5 of the failures)

Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Recurse.Get-ChildItem -Path -Recurse -Hidden
```powershell
Expected the value to have type [System.IO.FileInfo] or any of its subtypes, but got $null with type $null.
at <ScriptBlock>, /Users/runner/runners/2.169.1/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystemProviderExtended.Tests.ps1: line 175
175:             $result.Where({ $_.Name -eq ".filehidden3.txt" }) | Should -BeOfType System.IO.FileInfo
```
Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Recurse.Get-ChildItem -Path -Recurse -Attributes Hidden
```powershell
Expected the value to have type [System.IO.FileInfo] or any of its subtypes, but got $null with type $null.
at <ScriptBlock>, /Users/runner/runners/2.169.1/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystemProviderExtended.Tests.ps1: line 184
184:             $result.Where({ $_.Name -eq ".filehidden3.txt" }) | Should -BeOfType System.IO.FileInfo
```
Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Recurse.Get-ChildItem -Path -Recurse -Force
```powershell
Expected the value to have type [System.IO.FileInfo] or any of its subtypes, but got $null with type $null.
at <ScriptBlock>, /Users/runner/runners/2.169.1/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystemProviderExtended.Tests.ps1: line 193
193:             $result.Where({ $_.Name -eq ".filehidden3.txt" }) | Should -BeOfType System.IO.FileInfo
```
Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Recurse.Get-ChildItem -Path -Recurse -File -Hidden
```powershell
Expected the value to have type [System.IO.FileInfo] or any of its subtypes, but got $null with type $null.
at <ScriptBlock>, /Users/runner/runners/2.169.1/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystemProviderExtended.Tests.ps1: line 216
216:             $result.Where({ $_.Name -eq ".filehidden3.txt" }) | Should -BeOfType System.IO.FileInfo
```
Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Recurse -Name.Get-ChildItem -Path -Recurse -Name -Hidden
```powershell
Expected the value to have type [string] or any of its subtypes, but got $null with type $null.
at <ScriptBlock>, /Users/runner/runners/2.169.1/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystemProviderExtended.Tests.ps1: line 272
272:             $result.Where({ $_ -eq "subDir3$($DirSep).filehidden3.txt" }) | Should -BeOfType System.String
```